src/org/omegat/gui/dictionaries/DictionariesTextArea.java

```diff
@@ -257,8 +259,25 @@ protected void setFoundResult(final SourceTextEntry se, final List<DictionaryEnt
 
             displayedWords.add(de.getWord().toLowerCase());
             i++;
+            if (i % textBatch == 0) {
+                pos = pos + addText(pos, txt.toString());
+                txt = new StringBuilder();
+            }
+        }
+        addText(pos, txt.toString());
+    }
+
+    private int addText(final int pos, final String txt) {
```

Comment:
We are always appending to the end of the text, so I think it should be fast enough to simply call `document.getLength()` and always append at that position. Then you can remove the `pos` bookkeeping. The method could be called `appendText`.

Reply:
When using document.getLength(), I got index error.

Reply:
It seems working now.